src/org/omegat/util/FileUtil.java

```diff
@@ -472,9 +475,24 @@ public static List<String> buildRelativeFilesList(File rootDir, List<String> inc
             return p.toFile().isFile()
                     && FileUtil.checkFileInclude(root.relativize(p).toString(), includeMasks, excludeMasks);
         };
-        try (Stream<Path> stream = Files.find(root, Integer.MAX_VALUE, pred, FileVisitOption.FOLLOW_LINKS)) {
-            return stream.map(p -> root.relativize(p).toString().replace('\\', '/'))
-                    .sorted(StreamUtil.localeComparator(Function.identity())).collect(Collectors.toList());
+        int maxDepth = Integer.MAX_VALUE;
+        try (Stream<Path> stream = Files.find(root, maxDepth, pred, FileVisitOption.FOLLOW_LINKS)) {
+            List<String> result = new ArrayList<>();
+            try {
+                stream.forEach(p -> result.add(root.relativize(p).toString().replace('\\', '/')));
+            } catch (UncheckedIOException e) {
+                if (e.getCause() != null && e.getCause() instanceof FileSystemLoopException) {
+                    // source file folder has a looped symbolic links
+                    String p = ((FileSystemLoopException) e.getCause()).getFile();
+                    Core.getMainWindow().displayErrorRB(e.getCause(), "TF_LOAD_ERROR_SOURCE_LOOP_EXCEPTION", p);
+                    Log.logErrorRB(e, "TF_LOAD_WARN_SOURCE_LOOP_EXCEPTION", p);
+                    // stopping walking source folder and continue
+                } else {
+                    throw e; // propagate to caller
```

Comment:
It may be better we check an `UncheckedIOException` cause and throw a proper error  class.

Reply:
Bing search chat teach me... 
>According to the web results, UncheckedIOException in Java is a class that wraps an IOException with an unchecked exception. It is a subclass of RuntimeException and was introduced in Java 8.

So I think we can unwrap UncheckedIOException object using `getCause()`  and it will be a subclass of IOException.